pkg/network/nmstate.go

```diff
@@ -0,0 +1,40 @@
+package network
+
+import (
+	"os"
+	"path/filepath"
+	"reflect"
+
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/render"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+)
+
+func changeSafeNMState(prev, next *opv1alpha1.NetworkAddonsConfigSpec) []error {
+	if prev.NMState != nil && !reflect.DeepEqual(prev.NMState, next.NMState) {
+		return []error{errors.Errorf("cannot modify NMState state handler configuration once it is deployed")}
+	}
+	return nil
+}
+
+// renderNMState generates the manifests of NMState handler
+func renderNMState(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string, clusterInfo *ClusterInfo) ([]*unstructured.Unstructured, error) {
+	if conf.NMState == nil {
+		return nil, nil
+	}
+
+	// render the manifests on disk
+	data := render.MakeRenderData()
+	data.Data["NMStateStateHandlerImage"] = os.Getenv("NMSTATE_STATE_HANDLER_IMAGE")
```

Comment:
related to CI error 
```
status:
  conditions:
  - lastProbeTime: 2019-05-09T20:06:57Z
    lastTransitionTime: 2019-05-09T20:06:10Z
    message: 'failed to render: failed to render nmstate state handler manifests:
      error rendering manifests: failed to render manifest data/nmstate/002-nmstated.yaml:
      template: data/nmstate/002-nmstated.yaml:22:20: executing "data/nmstate/002-nmstated.yaml"
      at <.NMStateHandlerImage>: map has no entry for key "NMStateHandlerImage"'
    reason: FailedToRender
    status: "True"
    type: Failing
FAILED
```

can you please change here to 
```
NMStateStateHandlerImage -> NMStateHandlerImage
```

Reply:
I think we said we wanted it to be verbosely named?  unless the same image will be used for both controller and client?

Reply:
I've changed the template to be using the correct variable as defined in pkg/network/nmstate.go